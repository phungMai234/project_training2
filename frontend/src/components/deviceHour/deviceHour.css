.device-hour {
    margin: 20px 0px;
}

.custom-marker {
    width: 100px;
    height: 10px;
}