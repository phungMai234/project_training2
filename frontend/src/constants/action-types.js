const GET_DEVICE_TYPE = 'GET_DEVICE_TYPE';
const ERROR= "ERROR"
export {
  GET_DEVICE_TYPE,
  ERROR
}