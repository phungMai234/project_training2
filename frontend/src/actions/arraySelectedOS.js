import {GET_ARRAY_SELECTED_OS} from "../constants/action-types";

const getArraySelectedOSSuccess = (data) => ({
  type: GET_ARRAY_SELECTED_OS.SUCCESS,
  payload: data
})
const getArraySelectedOSError = (message) => ({
  type: GET_ARRAY_SELECTED_OS.ERROR,
  message: message
})
export const getArraySelectedOS = (arrayOS) => {
  return dispatch => {
    if (!arrayOS) {
      dispatch(getArraySelectedOSError("Select at least OS"));
    }
    dispatch(getArraySelectedOSSuccess(arrayOS))
  }
}
