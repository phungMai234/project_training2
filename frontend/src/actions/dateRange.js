import {GET_DATE_RANGE} from "../constants/action-types";

const getDateRangeSuccess = (data) => ({
  type: GET_DATE_RANGE.SUCCESS,
  payload: data
})
const getDateRangeError = (message) => ({
  type: GET_DATE_RANGE.ERROR,
  message: message
})
export const getDateRange = (dateRange) => {
  return dispatch => {
    if (!dateRange.startDate || !dateRange.endDate) {
      dispatch(getDateRangeError("Missing date range"));
    }
    dispatch(getDateRangeSuccess(dateRange))
  }
}
