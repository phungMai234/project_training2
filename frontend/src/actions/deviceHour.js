import {GET_HOUR} from "../constants/action-types";
import axios from "axios"

const getHourRequest = () => ({
  type: GET_HOUR.PENDING
})
const getHourSuccess = (data) => ({
  type: GET_HOUR.SUCCESS,
  payload: data
})
const getHourError = (message) => ({
  type: GET_HOUR.ERROR,
  message: message
})
export const getHour = () => {
  return dispatch => {
    dispatch(getHourRequest());
    return axios.get("http://localhost:3001/request/device_hour")
      .then(res => {
        console.log("res", res)
        return dispatch(getHourSuccess(res.data))
      })
      .catch(err => dispatch(getHourError(err.message)));
  }
}
