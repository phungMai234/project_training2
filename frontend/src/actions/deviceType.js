import {GET_DEVICE_TYPE} from "../constants/action-types";
<<<<<<< HEAD
import {getTypes} from "../api/deviceType-api";

export const  getDeviceType = async (dispatch) => {
    const res = await getTypes();
    return dispatch({
      type: GET_DEVICE_TYPE,
      res
    })
=======

export const getDeviceType = (types) =>{
    return {
        type: GET_DEVICE_TYPE,
        types: types
    }
>>>>>>> 3192a90e76a73f76f3377cd371d1f1101668d4e5
}
