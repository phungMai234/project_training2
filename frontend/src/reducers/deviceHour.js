import {GET_HOUR} from "../constants/action-types";

const initialState = {
  data: [],
  isFetching: false
}
export const deviceHourReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_HOUR.SUCCESS: {
      return {...state, isFetching: false, data: action.payload}
    }
    case GET_HOUR.ERROR: {
      return {...state, isFetching: false, errorMessage: action.message}
    }
    case GET_HOUR.PENDING: {
      return {...state, isFetching: true}
    }
    default:
      return state;
  }
}