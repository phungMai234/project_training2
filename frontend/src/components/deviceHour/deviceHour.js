import React, {useEffect} from "react";
import {connect} from "react-redux"
import "./deviceHour.css"
import HeatMap from "../heatMap/heatMap";
import BarChart from "../barChart/barChart";
import {Spinner} from "../spinner/spinner";
import {getHour} from "../../actions/deviceHour";

function DeviceHour(props) {
  const isFetching = props.isFetching;
  const dateRange = props.dateRange;

  useEffect(() => {
    const fetchData = () => {
      return props.getHour(dateRange);
    }
    fetchData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dateRange]);
  return (<div className="device-hour">
    <label className="text-secondary">Device by hour</label>
    {
      isFetching ? <Spinner/> :
        <div className="row">
          <div className="col-sm-9">
            <HeatMap/>
          </div>
          <div className="col-sm-3">
            <BarChart/>
          </div>
        </div>
    }
  </div>);

}

const mapStateToProps = state => ({
  isFetching: state.hour.isFetching,
  dateRange: state.dateRange.data
})
const mapDispatchToProps = dispatch => ({
  getHour: (dateRange) => dispatch(getHour(dateRange))
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviceHour);