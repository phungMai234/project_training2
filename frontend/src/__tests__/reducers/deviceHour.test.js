import {getHour} from "../../actions/deviceHour";
import {deviceHourReducer} from "../../reducers/deviceHour";
import {GET_HOUR} from "../../constants/action-types";

const initialState = {
  data: [],
  isFetching: false
}
describe("device types reducer", () => {
  it("should get device types to empty list", () => {
    expect(deviceHourReducer(initialState, getHour())).toMatchSnapshot();
  })
  it('should handle unknown actions', () => {
    expect(deviceHourReducer(undefined, {})).toEqual(initialState);
  });
  it("should handle loading time", () => {
    expect(deviceHourReducer(initialState, {type: GET_HOUR.PENDING})).toEqual({data: [], isFetching: true});
  })
  it('should handle error', (err = "error") => {
    expect(deviceHourReducer(initialState, {type: GET_HOUR.ERROR, message: err})).toEqual({
      data: [],
      isFetching: false,
      errorMessage: err
    });
  });
})