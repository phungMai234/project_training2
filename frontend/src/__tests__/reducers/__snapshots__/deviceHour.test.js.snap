// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`device types reducer should get device types to empty list 1`] = `
Object {
  "data": Array [],
  "isFetching": false,
}
`;
